cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LabelExpressionsTest.scala

```diff
@@ -95,5 +95,32 @@ class labelExpressionsTest extends DocumentingTest with QueryStatisticsTestSuppo
         resultTable()
       }
     }
+    section("Match with a nesting of label expressions for the node labels", "syntax-nesting-label") {
+      p("""A match with a nesting of label expressions for the node label returns the nodes for which the full expression is true""".stripMargin('#'))
+      query("""MATCH (n:(!A&!B)|C) RETURN n.name AS name""".stripMargin('#'),
+        ResultAssertions((r) => {
+          r.toList should equal(List(Map("name" -> "Charlie"), Map("name" -> "Eskil"), Map("name" -> "Frank"), Map("name" -> "George"), Map("name" -> "Henry")))
+        })) {
+        resultTable()
+      }
+    }
+    section("Match with label expressions in the predicate", "syntax-predicate-label") {
+      p("""A label expression can also be used in the predicate""".stripMargin('#'))
```

Comment:
I think we can reformulate this a bit (since it kind of is a predicate in the MATCH clause as well)

```suggestion
      p("""A label expression can also be used as a predicate in the WHERE clause""".stripMargin('#'))
```